tests/testthat/test-get_source_expressions.R

```diff
@@ -307,6 +307,26 @@ test_that("Syntax errors in Rmd or qmd don't choke lintr", {
   expect_silent(get_source_expressions(tmp))
 })
 
+test_that("Indented Rmd chunks don't cause spurious whitespace lints", {
+  tmp <- withr::local_tempfile(lines = c(
+    "* An enumeration item with code:",
+    "",
+    "  ```{r}",
+    '  "properly indented"',
+    "  ```",
+    "",
+    "# New section",
+    "",
+    "```{r unindented_chunk}",
+    '  "improperly indented"',
+    "```"
+  ))
+
+  parsed_lines <- get_source_expressions(tmp)$lines
+  expect_identical(parsed_lines[4L], '"properly indented"', ignore_attr = "names")
+  expect_identical(parsed_lines[10L], '  "improperly indented"', ignore_attr = "names")
```

Comment:
Can we also test that we don't destroy code?
````
  ```{r weird indent}
"no whitespace before"
  ```
````

Runs through `rmarkdown::render()` just fine.

Reply:
To be clear, I don't mind this creating (or not creating) spurious lints.
It just shouldn't create parse errors or the like.

Reply:
ah i really wish `rmarkdown::render()` would error there :)

Reply:
I'm tempted to punt on this. It's easy enough to do something like `gsub("^[ \t>]{1,N}", "", line)` where `N` is the indent column number, but then what about

       ```{r weirder_indent}
    "at column 1"
     "at column 2"
       ```

Ideally we'd lint (I think `indentation_linter()`?) on the second line, so actually we need to find the first column for actual code in each chunk and outdent that much (harder than current setup)

Reply:
actually in writing this comment, maybe I answered my own question :)

Reply:
Still a bit messy, but not horrible; done.